internal/controllers/clusterextension_controller.go

```diff
@@ -306,22 +307,32 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 // SetDeprecationStatus will set the appropriate deprecation statuses for a ClusterExtension
 // based on the provided bundle
 func SetDeprecationStatus(ext *ocv1alpha1.ClusterExtension, bundleName string, deprecation *declcfg.Deprecation) {
-	deprecations := map[string]declcfg.DeprecationEntry{}
+	deprecations := map[string][]declcfg.DeprecationEntry{}
+	channelSet := sets.New[string]()
+	if ext.Spec.Source.Catalog != nil {
+		for _, channel := range ext.Spec.Source.Catalog.Channels {
+			channelSet.Insert(channel)
+		}
+	}
 	if deprecation != nil {
 		for _, entry := range deprecation.Entries {
 			switch entry.Reference.Schema {
 			case declcfg.SchemaPackage:
-				deprecations[ocv1alpha1.TypePackageDeprecated] = entry
+				deprecations[ocv1alpha1.TypePackageDeprecated] = []declcfg.DeprecationEntry{entry}
 			case declcfg.SchemaChannel:
-				if ext.Spec.Source.Catalog.Channel != entry.Reference.Name {
-					continue
+				if channelSet.Has(entry.Reference.Name) {
+					if deps, ok := deprecations[ocv1alpha1.TypeChannelDeprecated]; ok {
+						deps = append(deps, entry)
+						deprecations[ocv1alpha1.TypeChannelDeprecated] = deps
+						continue
+					}
+					deprecations[ocv1alpha1.TypeChannelDeprecated] = []declcfg.DeprecationEntry{entry}
```

Comment:
```suggestion
					deprecations[ocv1alpha1.TypeChannelDeprecated] = append(deprecations[ocv1alpha1.TypeChannelDeprecated], entry)
```

Reply:
Does this handle the case where `deprecations[ocv1alpha1.TypeChannelDeprecated]` has not yet been populated (i.e a new `[]declfg.DeprecationEntry` is initialized if this is the first time we are adding this key/value pair)?

Reply:
Used go playground to verify it does - TIL!